### PR TITLE
Add small script to remove trailing whitespace; add to make fmt 

### PR DESCRIPTION
**Pull request by egibs**
### Background

We want a way to remove trailing whitespace using our existing `make fmt` flow. This PR adds a small OS-agnostic script to handle this via `make fmt`.

Inspired by #1109 -- we just want to avoid modifying so many files at one time to avoid potential downstream merge conflicts.

### Changes

* Adds `format.sh` to remove trailing whitespace from Python and YAML files

### Testing

* I tested this on both Linux and MacOS and all of the files were formatted as expected

### CODE REVIEW

**Review by kjihso — COMMENTED** [review]
Thanks for working on the trailing whitespace issue, perhaps influenced by #1109. 

Limiting this to *.py and *.yml doesn't match the markdown and other non-python/non-yaml files in #1109.

Also, I think that the regex for sed needs to be adjusted.

``[[:space]]*`` will match zero or more white space characters. What is intended is to match one or more, thus the metacharacter ``+`` should be used or perhaps ``\{1,`` depending on BRE vs ERE (yes, I see -E used with sed. Yay!). Also, one will have to make sure these seds agree (perhaps already done) that the metacharacter will always be bare instead of preceded by ``\`` to be considered a metacharacter due to Darwin sed (BSD heritage) not being GNU Sed (Linux).

The substitution should not be made on lines in a file that have zero trailling whitespace characters as this would cause lines that don't have trailing whitespace to be acted on, which is superfluous.

I love regexes. I hate their myriad of implementations and how they conflict (and I'm probably not getting all the details above correct--it's easy to miss some between the conflicting implementations, thus the use of a perl one-liner below).

The following was used to fix this when I generated #1109 (yes, one can find -regex and use sed, I just loathe both):

find .[a-z]* * -type f -print0 | grep -zZv '^.git/' | xargs -0 perl -pi.bak -e 's/\s+$//'

In any case, thank you for considering #1109 (assuming that's what's happened) and making adjustments. We will look forward to less merge fun (lol) for us in the future with trailing whitespace style in sync.

**Comment by egibs** [comment]
> The following was used to fix this when I generated #1109 (yes, one can find -regex and use sed, I just loathe both):
> 
> find .[a-z]* * -type f -print0 | grep -zZv '^.git/' | xargs -0 perl -pi.bak -e 's/\s+$//'
> 
> In any case, thank you for considering #1109 (assuming that's what's happened) and making adjustments. We will look forward to less merge fun (lol) for us in the future with trailing whitespace style in sync.

Ah, good to know! I think it would make sense to add this since it's what you used originally.

We just wanted a less drastic way of enabling this behavior instead of formatting every file at one time. That said, we appreciate the effort because it's good hygiene and often overlooked. I'll reference your PR from this PR as well.

File: format.sh [hunk 1]
```diff
@@ -1,12 +1,4 @@
...
 
-OS=$(uname)
-for extension in "*.py" "*.yml"; do
-    case $OS in
-        Darwin) find . -type f -not -path '.git' -iname $extension -print0 | xargs -0 sed -i '' -E "s/[[:space:]]*\$//";;
-        Linux) find . -type f -not -path '.git' -iname $extension -print0 | xargs -0 sed -i -E "s/[[:space:]]*\$//";;
-        *) echo "Unsupported OS: $OSTYPE"; exit 1;;
-    esac
-done
+find .[a-z]* * -type f -print0 | grep -zZv '^.git/' | xargs -0 perl -pi -e 's/[ \t]+$//'
```
[the remark this answers is not in the document]
Comment (egibs) [reply format.sh]: Makes sense -- addressed in [`5545344` (#1119)](https://github.com/panther-labs/panther-analysis/pull/1119/commits/5545344634ec023f69a7f6c82201f7c2f97f1ce3).

**Review by kjihso — APPROVED** [review]

**Comment by le4ker** [comment]
@kjihso we are about to add prettier for YAML and Markdown formatting at #1153 . Do you want to checkout the branch and run a `make fmt` to check it out?

**Comment by kjihso** [comment]
Thanks @le4ker . Tested that change in our repo and it LGTM.

**Pull request closed without merging**